flipt-engine-ffi/build.sh

```diff
@@ -1,13 +1,46 @@
 #!/bin/bash
 
-cargo build -p flipt-engine-ffi --release --target=aarch64-apple-ios-sim
-cargo build -p flipt-engine-ffi --release --target=aarch64-apple-ios
-cargo build -p flipt-engine-ffi --release --target=aarch64-apple-darwin
+# Check if SDK argument is provided
+if [ $# -ne 1 ]; then
+  echo "Error: SDK argument is required"
+  echo "Usage: $0 [swift|android]"
+  exit 1
+fi
 
-rm -rf ../flipt-client-swift/Sources/FliptEngineFFI.xcframework
+SDK=$1
 
-xcodebuild -create-xcframework \
-        -library ../target/aarch64-apple-ios-sim/release/libfliptengine.a -headers ./include/ \
-        -library ../target/aarch64-apple-ios/release/libfliptengine.a -headers ./include/ \
-        -library ../target/aarch64-apple-darwin/release/libfliptengine.a -headers ./include/ \
-        -output "../flipt-client-swift/Sources/FliptEngineFFI.xcframework"
+case $SDK in
+"swift")
+  rustup target add aarch64-apple-ios x86_64-apple-ios aarch64-apple-ios-sim
+
+  cargo build -p flipt-engine-ffi --release --target=aarch64-apple-ios-sim
+  cargo build -p flipt-engine-ffi --release --target=aarch64-apple-ios
+  cargo build -p flipt-engine-ffi --release --target=aarch64-apple-darwin
+
+  rm -rf ../flipt-client-swift/Sources/FliptEngineFFI.xcframework
+
+  xcodebuild -create-xcframework \
+    -library ../target/aarch64-apple-ios-sim/release/libfliptengine.a -headers ./include/ \
+    -library ../target/aarch64-apple-ios/release/libfliptengine.a -headers ./include/ \
+    -library ../target/aarch64-apple-darwin/release/libfliptengine.a -headers ./include/ \
+    -output "../flipt-client-swift/Sources/FliptEngineFFI.xcframework"
+  ;;
+
+"android")
+  rustup target add x86_64-linux-android aarch64-linux-android
+
+  cargo build -p flipt-engine-ffi --release --target=x86_64-linux-android
+  cargo build -p flipt-engine-ffi --release --target=aarch64-linux-android
+  mkdir -p ../flipt-client-kotlin-android/src/main/cpp/libs/x86_64
+  mkdir -p ../flipt-client-kotlin-android/src/main/cpp/libs/arm64-v8a
+  cp ../target/x86_64-linux-android/release/libfliptengine.a ../flipt-client-kotlin-android/src/main/cpp/libs/x86_64/libfliptengine.a
+  cp ../target/aarch64-linux-android/release/deps/libfliptengine.so ../flipt-client-kotlin-android/src/main/cpp/libs/arm64-v8a/libfliptengine.a
```

Comment:
fwiw I dont think we need the arm libraries as this test should only run on x86 linux on GitHub actions. 

also why `./target/aarch64-linux-android/release/deps/libfliptengine.so` ?

Reply:
gradle wants to have all the assets :( 

I  wish I could use outputs from build engine ffi action as input for this. Don't know if it's possible.

Reply:
ah, im curious about two things:

1. why its in the `deps` subfolder
2. why its a dynamic `.so` ?